### PR TITLE
Add google-java-format and reformat

### PR DESCRIPTION
**Pull request by JakeWharton**

### CODE REVIEW

File: retrofit-adapters/guava/src/main/java/retrofit2/adapter/guava/GuavaCallAdapterFactory.java [hunk 1]
```diff
@@ -89,26 +92,31 @@ private GuavaCallAdapterFactory() {
       this.responseType = responseType;
     }
 
-    @Override public Type responseType() {
+    @Override
+    public Type responseType() {
```
Comment (oldergod) [inline retrofit-adapters/guava/src/main/java/retrofit2/adapter/guava/GuavaCallAdapterFactory.java:96]:
```suggestion
    @Override public Type responseType() {
```

Comment (JakeWharton) [reply retrofit-adapters/guava/src/main/java/retrofit2/adapter/guava/GuavaCallAdapterFactory.java:96]: https://github.com/google/google-java-format/issues/342

File: build.gradle [hunk 2]
```diff
@@ -103,6 +103,28 @@ subprojects {
...
       }
     }
+
+    project.apply plugin: 'com.github.sherter.google-java-format'
+    googleJavaFormat {
+      toolVersion = '1.7'
+
+      // By default, the GJF plugin includes all Java folders inside the project directory. This
+      // does not work well with nested projects, especially when you want to exclude them.
+      source = sourceSets*.allJava
```
Comment (swankjesse) [inline build.gradle:113]: I’m trying to imagine what this syntax does

File: retrofit-adapters/guava/src/main/java/retrofit2/adapter/guava/GuavaCallAdapterFactory.java [hunk 3]
```diff
@@ -75,8 +78,8 @@ private GuavaCallAdapterFactory() {
 
     // Generic type is Response<T>. Extract T and create the Response version of the adapter.
     if (!(innerType instanceof ParameterizedType)) {
-      throw new IllegalStateException("Response must be parameterized"
-          + " as Response<Foo> or Response<? extends Foo>");
+      throw new IllegalStateException(
+          "Response must be parameterized" + " as Response<Foo> or Response<? extends Foo>");
```
Comment (swankjesse) [inline retrofit-adapters/guava/src/main/java/retrofit2/adapter/guava/GuavaCallAdapterFactory.java:82]: supernit: the + is now weird

File: retrofit-adapters/guava/src/test/java/retrofit2/adapter/guava/GuavaCallAdapterFactoryTest.java [hunk 4]
```diff
@@ -39,15 +39,18 @@
...
-  @Before public void setUp() {
-    retrofit = new Retrofit.Builder()
-        .baseUrl(server.url("/"))
-        .addConverterFactory(new StringConverterFactory())
-        .addCallAdapterFactory(factory)
-        .build();
+  @Before
+  public void setUp() {
+    retrofit =
+        new Retrofit.Builder()
```
Comment (swankjesse) [inline retrofit-adapters/guava/src/test/java/retrofit2/adapter/guava/GuavaCallAdapterFactoryTest.java:45]: oooh, this formatter has bad taste

File: retrofit-adapters/rxjava/src/main/java/retrofit2/adapter/rxjava/RxJavaCallAdapter.java [hunk 5]
```diff
@@ -44,14 +50,15 @@
...
   }
 
-  @Override public Object adapt(Call<R> call) {
-    OnSubscribe<Response<R>> callFunc = isAsync
-        ? new CallEnqueueOnSubscribe<>(call)
-        : new CallExecuteOnSubscribe<>(call);
+  @Override
+  public Object adapt(Call<R> call) {
+    OnSubscribe<Response<R>> callFunc =
+        isAsync ? new CallEnqueueOnSubscribe<>(call) : new CallExecuteOnSubscribe<>(call);
```
Comment (swankjesse) [inline retrofit-adapters/rxjava/src/main/java/retrofit2/adapter/rxjava/RxJavaCallAdapter.java:61]: ?: on their own line is so good though

File: retrofit-adapters/rxjava2/src/main/java/retrofit2/adapter/rxjava2/RxJava2CallAdapterFactory.java [hunk 6]
```diff
@@ -111,25 +113,30 @@ private RxJava2CallAdapterFactory(@Nullable Scheduler scheduler, boolean isAsync
...
-          : isSingle ? "Single"
-          : isMaybe ? "Maybe" : "Observable";
-      throw new IllegalStateException(name + " return type must be parameterized"
-          + " as " + name + "<Foo> or " + name + "<? extends Foo>");
+      String name =
+          isFlowable ? "Flowable" : isSingle ? "Single" : isMaybe ? "Maybe" : "Observable";
+      throw new IllegalStateException(
+          name
+              + " return type must be parameterized"
+              + " as "
```
Comment (swankjesse) [inline retrofit-adapters/rxjava2/src/main/java/retrofit2/adapter/rxjava2/RxJava2CallAdapterFactory.java:121]: ick

File: retrofit-converters/java8/src/test/java/retrofit/converter/java8/Java8OptionalConverterFactoryTest.java [hunk 7]
```diff
@@ -31,57 +33,67 @@
...
-          }
-        })
-        .addConverterFactory(Java8OptionalConverterFactory.create())
-        .build();
+    Retrofit retrofit =
+        new Retrofit.Builder()
+            .baseUrl(server.url("/"))
+            .addConverterFactory(
+                new Converter.Factory() {
+                  @Nullable
```
Comment (swankjesse) [inline retrofit-converters/java8/src/test/java/retrofit/converter/java8/Java8OptionalConverterFactoryTest.java:85]: oooh I think this one is wrong

Comment (swankjesse) [reply retrofit-converters/java8/src/test/java/retrofit/converter/java8/Java8OptionalConverterFactoryTest.java:85]: weird that it’s different from the other `@Nullable` ones. Must retain annotation order strongly.

File: retrofit-converters/jaxb/src/test/java/retrofit2/converter/jaxb/JaxbConverterFactoryTest.java [hunk 8]
```diff
@@ -130,35 +136,42 @@
...
-            + "<phone_number type=\"FAX\">"
-            + "<number>867-5309</number>"
-            + "</phone_number>"
-            + "</contact>"));
+  @Test
+  public void unrecognizedField() throws Exception {
+    server.enqueue(
+        new MockResponse()
+            .setBody(
+                ""
```
Comment (swankjesse) [inline retrofit-converters/jaxb/src/test/java/retrofit2/converter/jaxb/JaxbConverterFactoryTest.java:144]: another questionable one

File: retrofit/src/main/java/retrofit2/RequestFactory.java [hunk 9]
```diff
@@ -99,12 +99,24 @@ static RequestFactory parseAnnotations(Retrofit retrofit, Method method) {
...
+              + argumentCount
+              + ") doesn't match expected count ("
+              + handlers.length
+              + ")");
     }
 
-    RequestBuilder requestBuilder = new RequestBuilder(httpMethod, baseUrl, relativeUrl,
-        headers, contentType, hasBody, isFormEncoded, isMultipart);
+    RequestBuilder requestBuilder =
+        new RequestBuilder(
```
Comment (swankjesse) [inline retrofit/src/main/java/retrofit2/RequestFactory.java:111]: I find this kind of change quite interesting

File: retrofit/src/test/java/retrofit2/RetrofitTest.java [hunk 10]
```diff
@@ -793,43 +908,41 @@
...
-        .callFactory(callFactory)
-        .build();
+  @Test
+  public void callFactoryPropagated() {
+    okhttp3.Call.Factory callFactory =
+        request -> {
+          throw new AssertionError();
+        };
+    Retrofit retrofit =
+        new Retrofit.Builder().baseUrl("http://example.com/").callFactory(callFactory).build();
```
Comment (swankjesse) [inline retrofit/src/test/java/retrofit2/RetrofitTest.java:924]: another weird one

File: retrofit-adapters/guava/src/main/java/retrofit2/adapter/guava/GuavaCallAdapterFactory.java (hunk 3 above)
Comment (JakeWharton) [reply retrofit-adapters/guava/src/main/java/retrofit2/adapter/guava/GuavaCallAdapterFactory.java:82]: yeah I noticed this as well. I'll check and file bugs for some of these.

**Pull request merged**